design/milestone_1_2_design_doc.md

```diff
@@ -350,25 +350,27 @@ For a detailed breakdown of the decision process see [here](batch_retrieval_desi
+|      | Solution                               | Elaboration                                                  | Pros                                                         | Cons                                                         | Effort estimation |
+| ---- | -------------------------------------- | ------------------------------------------------------------ | ------------------------------------------------------------ | ------------------------------------------------------------ | ----------------- |
+| 1    | Client-side polling for secrets        | Every configurable amount of time, contact Conjur API server.<br />Possible solutions can be:<br />1. Retrieve all secrets<br />2. Check if there's something new for the requesting host<br />3. Retrieve secrets that were changed since last request | - Straight-forward<br />- Simple<br />- No additional requirements needed | - Regularly sends API calls even if nothing has changed<br />- Introduces load on the server to hanlde many secrets providers | 15 days           |
+| 2    | Event-driven mechanism using pub-sub   | Secrets providers subscribes to relevant events in the server. Server publishes a secret-changed event if a secret was changed | - Efficient, client calls the server only when needed<br />- Allows serving many secrets providers without too much load on the server | - Maintains active connection in the background<br />- Introduces Events mechanism that does not exist<br />- Complex<br />- Need to open ports for allowing connection | 40 days           |
+| 3    | Event-driven mechanism using web hooks | Secrets providers registers a URL for calling back when a secret has changed. | - Efficient, client calls the server only when needed<br />- No active connection is held in the background. A connection is made by the server only when needed. | - Complex<br />- Requires allowing connection from server to secrets providers<br />- Follower cannot keep callback URL in the DB, so no guarantee it will be used when needed<br />- Introduces Events mechanism that does not exist | 30 days           |
 
-(*) As a possible fallback, we can provide the user with a warning that the endpoint feature is not available in their Conjur version and fail back to Solution #1.
+We decided to go with option No. 1.
+Low level design of the solution will be introduced in a separate document.
 
-#### Concerns
+#### Customer experience
 
-*Load on Server*
+This solution does not introduce any additional requirements for the solution besides using `Deployment` instead of a `Job`.
```

Comment:
> any additional requirements

what do you mean by additional requirements? do you mean any addition user intervention?  as in the customer doesn't have to perform any additional tasks?